server/base/src/main/java/org/apache/accumulo/server/ServiceEnvironmentImpl.java

```diff
@@ -78,6 +79,22 @@ public String get(String key) {
       }
     }
 
+    @Override
+    public Map<String,String> getWithPrefix(String prefix) {
+      if (Property.isValidPropertyPrefix(prefix)) {
+        Property propertyPrefix = Property.getPropertyByKey(prefix);
+        return acfg.getAllPropertiesWithPrefix(propertyPrefix);
+      } else {
+        Map<String,String> properties = new HashMap<>();
+        for (Entry<String,String> prop : acfg) {
+          if (prop.getKey().startsWith(prefix)) {
+            properties.put(prop.getKey(), prop.getValue());
+          }
+        }
+        return properties;
+      }
```

Comment:
This can be more succinctly written with streams:
```suggestion
        return StreamSupport.stream(acfg.spliterator(), false)
            .filter(prop -> prop.getKey().startsWith(prefix))
            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
```

Reply:
Fixed